server/app/services/program/ProgramServiceImpl.java

```diff
@@ -765,6 +765,50 @@ public ProgramDefinition setProgramQuestionDefinitionOptionality(
     }
   }
 
+  @Override
+  @Transactional
+  public ProgramDefinition setProgramQuestionDefinitionAddressCorrectionEnabled(
+      long programId,
+      long blockDefinitionId,
+      long questionDefinitionId,
+      boolean addressCorrectionEnabled)
+      throws ProgramNotFoundException, ProgramBlockDefinitionNotFoundException,
+          ProgramQuestionDefinitionNotFoundException {
+    ProgramDefinition programDefinition = getProgramDefinition(programId);
+    BlockDefinition blockDefinition = programDefinition.getBlockDefinition(blockDefinitionId);
+
+    if (!blockDefinition.programQuestionDefinitions().stream()
+        .anyMatch(pqd -> pqd.id() == questionDefinitionId)) {
+      throw new ProgramQuestionDefinitionNotFoundException(
+          programId, blockDefinitionId, questionDefinitionId);
+    }
+
+    if (!blockDefinition.hasAddress()) {
+      throw new RuntimeException(
+          "Unexpected error: updating a non address question with address correction enabled");
+    }
+
+    ImmutableList<ProgramQuestionDefinition> programQuestionDefinitions =
+        blockDefinition.programQuestionDefinitions().stream()
+            .map(
+                pqd ->
+                    pqd.id() == questionDefinitionId
+                        ? pqd.setAddressCorrectionEnabled(addressCorrectionEnabled)
+                        : pqd)
+            .collect(ImmutableList.toImmutableList());
+
+    try {
+      return updateProgramDefinitionWithBlockDefinition(
+          programDefinition,
+          blockDefinition.toBuilder()
+              .setProgramQuestionDefinitions(programQuestionDefinitions)
+              .build());
+    } catch (IllegalPredicateOrderingException e) {
+      throw new RuntimeException(
+          "Unexpected error: updating this question invalidated a block condition");
+    }
```

Comment:
suggestion (blocking): wrap the original error

Reply:
updated to wrap the original error